src/libstd/sys/cloudabi/condvar.rs

```diff
@@ -131,14 +136,22 @@ impl Condvar {
                 ..mem::zeroed()
             },
         ];
-        let mut events: [abi::event; 2] = mem::uninitialized();
-        let mut nevents: usize = mem::uninitialized();
-        let ret = abi::poll(subscriptions.as_ptr(), events.as_mut_ptr(), 2, &mut nevents);
+        let mut events: [mem::MaybeUninit<abi::event>; 2] = [mem::MaybeUninit::uninit(); 2];
+        let events_slice = slice::from_raw_parts_mut(MaybeUninit::first_ptr_mut(&mut events), 2);
+        let mut nevents: mem::MaybeUninit<usize> = mem::MaybeUninit::uninit();
+        let ret = abi::poll(
+            subscriptions.as_ptr(),
+            events_slice,
```

Comment:
This still seems like it would not compile. `poll` takes a `*mut events` as its second argument, not a `&mut [events]`: https://github.com/rust-lang/rust/blob/master/src/libstd/sys/cloudabi/abi/cloudabi.rs#L2646

Reply:
Strangely enough still compiles on my end, even on a full build and test run (and looks to have compiled on the checks as well) but you're right on the type mismatch. Should be fixed now.